test/index.js

```diff
@@ -1026,23 +1025,26 @@ describe('fs', function() {
 
   it('watch()', function() {
     var watcher;
+    var target = pathFn.join(tmpDir, 'test.txt');
 
     return fs.watch(tmpDir).then(function(watcher_) {
       watcher = watcher_;
 
       return new Promise(function(resolve, reject) {
-        var path = pathFn.join(tmpDir, 'test.txt');
-
         watcher.on('add', function(path_) {
-          path_.should.eql(path);
+          try {
+            path_.should.eql(target);
+          } catch (err) {
+            reject(err);
+          }
           resolve();
         });
 
-        fs.writeFile(path, 'test').catch(reject);
+        fs.writeFile(target, 'test').catch(reject);
       });
     }).finally(function() {
       if (watcher) watcher.close();
-    });
```

Comment:
Can `return fs.unlink(target);` be used in `finally` block? It can simplify the promise chain.